server/base/src/main/java/org/apache/accumulo/server/security/handler/ZKSecurityTool.java

```diff
@@ -99,6 +102,24 @@ public static boolean checkPass(byte[] password, byte[] zkData) {
     return saltedHash; // contains salt+hash(password+salt)
   }
 
+  public static byte[] createPass(byte[] password) throws AccumuloException {
+    // we rely on default algorithm and hash length (SHA-512 and 8 byte)
+    String cryptHash = Crypt.crypt(password);
+    return cryptHash.getBytes(CRYPT_CHARSET);
+  }
+
+  public static boolean checkCryptPass(byte[] password, byte[] zkData) {
+    String zkDataString = new String(zkData, CRYPT_CHARSET);
+    String cryptHash;
+    try {
+      cryptHash = Crypt.crypt(password, zkDataString);
+    } catch (IllegalArgumentException e) {
+      log.error("Unrecognized hash format", e);
+      return false;
+    }
+    return MessageDigest.isEqual(zkData, cryptHash.getBytes(CRYPT_CHARSET));
```

Comment:
What's the advantage of using `MessageDigest.isEqual`, vs. comparing as Strings or comparing as the UTF-8 byte-array?

Reply:
String comparisons is not secure against timing attacks, `MessageDigest.isEqual` is.